dvc/tree/azure.py

```diff
@@ -36,9 +37,62 @@ def __init__(self, repo, config):
             container = self._az_config.get("storage", "container_name", None)
             self.path_info = self.PATH_CLS(f"azure://{container}")
 
-        self._conn_str = config.get(
-            "connection_string"
-        ) or self._az_config.get("storage", "connection_string", None)
+        self._get_string_token_credential(config)
+
+        if not (self._conn_str or self._credential):
+            from azure.identity import (
+                AzureCliCredential,
+                ChainedTokenCredential,
+            )
+
+            account_name = config.get("account_name")
+            self._account_url = f"https://{account_name}.blob.core.windows.net"
+            if not account_name:
+                logging.warning(
+                    "You must enter an account name for this auth flow"
+                )
```

Comment:
Should this be an error with a raised exception?

Reply:
Raising an exception broke a number of tests, one of which was `dvc\tests\unit\dependency\test_azure.py` because somewhere in the execution, a config object is created, one which I can't get a handle on, so the exception is raised and the test fails... this way is more in keeping with the previous behaviour where the user will just fail to authenticate against the remote and there will be logging if they wish to find more information about why